### ISSUE TITLE
Fix the API around capture* and handle callback correctly

### ISSUE THREAD

**Issue by mattrobenolt** [getsentry/raven-node#213]
Due to legacy reasons, `capture*`'s callback was to get the eventId.

So the current API is:

```javascript
client.captureError(new Error(), function(result) {
  console.log(client.getIdent(result));
});
```

This entire pattern is basically deprecated and unnecessary now for Node.

Ideally, we'd have a pattern like this now:

```javascript
var eventId = client.captureError(new Error(), function(err) {
  console.log(err);
});
```

Where the callback now indicates when the event has been sent or failed to send. And there's no reason now that we can't return the event id immediately.

This should be done for a 1.0 imo since this is a major breaking API change, but for the better.

**Comment by LewisJEllis** [comment]
Yea, I had realized we're capable of just immediately return the event id, but hadn't thought about the API/timing/related issues as much. Looking at them now I think this is the right move and I agree with the pattern you set out. It'll fit well with the other API changes happening in 1.0.0

Related: #200 #125 #196 #86

**Comment by LewisJEllis** [comment]
Addressed by #217.
